File: sdks/php/VaultClient.php
<?php

class VaultClient {
    private $baseUrl;
    private $apiKey;
    private $wallet;

    public function __construct($baseUrl, $apiKey, $wallet) {
        $this->baseUrl = rtrim($baseUrl, '/');
        $this->apiKey = $apiKey;
        $this->wallet = $wallet;
    }

    public function listFiles() {
        return $this->request('GET', '/api/files');
    }

    public function upload($filePath, $folderId = 0) {
        $curl = curl_init();
        curl_setopt_array($curl, [
            CURLOPT_URL => "{$this->baseUrl}/api/upload?folder_id={$folderId}",
            CURLOPT_POST => true,
            CURLOPT_POSTFIELDS => ['file' => new CURLFile($filePath)],
            CURLOPT_HTTPHEADER => $this->getHeaders(),
            CURLOPT_RETURNTRANSFER => true
        ]);
        $resp = curl_exec($curl);
        curl_close($curl);
        return json_decode($resp, true);
    }

    public function createFolder($name, $parentId = 0) {
        $curl = curl_init("{$this->baseUrl}/api/folders");
        curl_setopt($curl, CURLOPT_POST, true);
        curl_setopt($curl, CURLOPT_POSTFIELDS, json_encode(['name' => $name, 'parent_id' => $parentId]));
        curl_setopt($curl, CURLOPT_HTTPHEADER, array_merge($this->getHeaders(), ["Content-Type: application/json"]));
        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
        $resp = curl_exec($curl);
        curl_close($curl);
        return json_decode($resp, true);
    }

    public function listFolders() {
        return $this->request('GET', '/api/folders');
    }

    public function listVersions($fileId) {
        return $this->request('GET', "/api/files/{$fileId}/versions");
    }

    public function restoreVersion($fileId, $versionId) {
        return $this->request('POST', "/api/files/{$fileId}/restore/{$versionId}");
    }

    private function request($method, $path) {
        $curl = curl_init("{$this->baseUrl}{$path}");
        curl_setopt($curl, CURLOPT_CUSTOMREQUEST, $method);
        curl_setopt($curl, CURLOPT_HTTPHEADER, $this->getHeaders());
        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
        $resp = curl_exec($curl);
        curl_close($curl);
        return json_decode($resp, true);
    }

    private function getHeaders() {
        return [
            "Authorization: Bearer {$this->apiKey}",
            "X-Wallet-Address: {$this->wallet}"
        ];
    }
}

File: sdks/csharp/VaultClient.cs
using System;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Threading.Tasks;
using System.Collections.Generic;
using System.Text.Json;
using System.IO;

namespace Vault;

public class VaultClient {
    private readonly HttpClient _client;
    private readonly string _baseUrl;
    private readonly string _apiKey;
    private readonly string _wallet;

    public VaultClient(string baseUrl, string apiKey, string wallet) {
        _baseUrl = baseUrl.TrimEnd('/');
        _apiKey = apiKey;
        _wallet = wallet;
        _client = new HttpClient();
        _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiKey);
        _client.DefaultRequestHeaders.Add("X-Wallet-Address", wallet);
    }

    public async Task<string> ListFilesAsync() {
        var response = await _client.GetAsync($"{_baseUrl}/api/files");
        response.EnsureSuccessStatusCode();
        return await response.Content.ReadAsStringAsync();
    }

    public async Task<int> UploadAsync(string filePath, int folderId = 0) {
        using var form = new MultipartFormDataContent();
        using var fileContent = new ByteArrayContent(await File.ReadAllBytesAsync(filePath));
        fileContent.Headers.ContentType = MediaTypeHeaderValue.Parse("application/octet-stream");
        form.Add(fileContent, "file", Path.GetFileName(filePath));

        var response = await _client.PostAsync($"{_baseUrl}/api/upload?folder_id={folderId}", form);
        response.EnsureSuccessStatusCode();
        var json = await response.Content.ReadAsStringAsync();
        var doc = JsonDocument.Parse(json);
        return doc.RootElement.GetProperty("id").GetInt32();
    }

    public async Task<int> CreateFolderAsync(string name, int parentId = 0) {
        var content = new StringContent(JsonSerializer.Serialize(new { name = name, parent_id = parentId }), System.Text.Encoding.UTF8, "application/json");
        var response = await _client.PostAsync($"{_baseUrl}/api/folders", content);
        response.EnsureSuccessStatusCode();
        var json = await response.Content.ReadAsStringAsync();
        var doc = JsonDocument.Parse(json);
        return doc.RootElement.GetProperty("id").GetInt32();
    }

    public async Task<string> ListFoldersAsync() {
        var response = await _client.GetAsync($"{_baseUrl}/api/folders");
        response.EnsureSuccessStatusCode();
        return await response.Content.ReadAsStringAsync();
    }

    public async Task<string> ListVersionsAsync(int fileId) {
        var response = await _client.GetAsync($"{_baseUrl}/api/files/{fileId}/versions");
        response.EnsureSuccessStatusCode();
        return await response.Content.ReadAsStringAsync();
    }

    public async Task RestoreVersionAsync(int fileId, int versionId) {
        var response = await _client.PostAsync($"{_baseUrl}/api/files/{fileId}/restore/{versionId}", null);
        response.EnsureSuccessStatusCode();
    }
}
